homeassistant/components/qnap/coordinator.py

```diff
@@ -56,4 +56,7 @@ def _sync_update(self) -> dict[str, dict[str, Any]]:
 
     async def _async_update_data(self) -> dict[str, dict[str, Any]]:
         """Get the latest data from the Qnap API."""
-        return await self.hass.async_add_executor_job(self._sync_update)
+        result = await self.hass.async_add_executor_job(self._sync_update)
+        if not result:
```

Comment:
When there is a problem, what exactly happens? I just took a quick look and saw that it would return None.

With the current implementation a result when failing would be
```python
{
   "system_stats": None,
   "system_health": None,
   "smart_drive_health": None,
   "volumes": None,
   "bandwidth": None,
}
```
and this object is not falsey, so `not result` would return `False` instead of `True`.

When this situation happens, do all checks return None? Are there partial fails (so only 1 of 5 fails)?

Reply:
That is an excellent catch! In this case “system_stats” is the tell-tale. If that returns None then everything else will return None.

Reply:
Then `if not result.get("system_stats")` would do I think?

Reply:
This did not work

Reply:
Can I do "if system_stats == None?

Reply:
Could you write it in an example? Don't know how to picture this haha